arangod/Cluster/ClusterRepairDistributeShardsLike.h

```diff
@@ -0,0 +1,258 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2018 ArangoDB GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Tobias Gödderz
+////////////////////////////////////////////////////////////////////////////////
+
+#ifndef ARANGODB3_CLUSTERREPAIRDISTRIBUTESHARDSLIKE_H
+#define ARANGODB3_CLUSTERREPAIRDISTRIBUTESHARDSLIKE_H
+
+#include <arangod/Agency/AgencyComm.h>
+#include <velocypack/Compare.h>
+#include <velocypack/Slice.h>
+#include <velocypack/velocypack-common.h>
+
+#include <boost/optional.hpp>
+#include <boost/variant.hpp>
+
+#include "ClusterInfo.h"
+#include "ClusterRepairOperations.h"
+
+namespace arangodb {
+namespace velocypack {
+class Slice;
+template <typename T>
+class Buffer;
+}
+
+namespace cluster_repairs {
+
+using DBServers = std::vector<ServerID>;
+using VPackBufferPtr = std::shared_ptr<velocypack::Buffer<uint8_t>>;
+
+template <typename T, std::size_t N>
+std::ostream& operator<<(std::ostream& stream, std::array<T, N> array) {
+  stream << "std::array<" << typeid(T).name() << "> { ";
+  std::for_each(array.begin(), array.end(),
+                [&stream](T const& val) { stream << val << ", "; });
+
+  stream << "}";
+
+  return stream;
+}
+
+inline std::ostream& operator<<(std::ostream& stream,
+                                VPackBufferPtr const& vpack) {
+  return stream << "std::shared_ptr<VPackBuffer> { "
+                << velocypack::Slice(vpack->data()).toJson() << " "
+                << "}";
+}
+
+template <typename T>
+class ResultT : public arangodb::Result {
```

Comment:
Can we move this to a separate file in `lib/Basic/` this is useful in many places i think